risc0/zkvm/src/prove/exec.rs

```diff
@@ -802,23 +534,7 @@ impl<'a, H: HostHandler> MachineContext<'a, H> {
         }
     }
 
-    fn syscall_init(&mut self, cycle: usize) -> Result<()> {
-        assert!(self.syscall_out_data.is_empty());
-        let to_guest_words = self.memory.load_register(REG_A1);
-
-        let name_ptr = self.memory.load_register(REG_A2);
-
-        let name = self.memory.load_string(name_ptr)?;
-        debug!("SYS_IO[{cycle}] Guest requests {to_guest_words} words back");
-        let mut to_guest_buf = vec![0u32; to_guest_words as usize];
-
-        self.memory.cur_cycle = cycle;
-        self.syscall_out_regs = self
-            .handler
-            .on_txrx(&self.memory, &name, &mut to_guest_buf)?;
-        trace!("SYS_IO[{cycle}] (a0, a1): {:?}", self.syscall_out_regs);
-        trace!("SYS_IO[{cycle}] data sent to guest: {to_guest_buf:?}");
-        self.syscall_out_data = to_guest_buf.into();
+    fn syscall_init(&mut self, _cycle: usize) -> Result<()> {
```

Comment:
Can this function be removed all together? The only other reference I see is on line 229

Reply:
Yes, sort of. Right now the circuit expects this extern to exist, but we might be able to remove it from the circuit in the future. Leaving it in for now as a note that it's still used by the circuit.